meson.build

```diff
@@ -231,10 +231,10 @@ endif
 subdir('Common/src')
 # compile SU2_CFD executable
 subdir('SU2_CFD/src')
-# compile SU2_DOT executable
-subdir('SU2_DOT/src')
 # compile SU2_DEF executable
 subdir('SU2_DEF/src')
+# compile SU2_DOT executable
```

Comment:
why did this move?

Reply:
Because `SU2_DOT` is now dependent on `CDiscAdjDeformationDriver (SU2_DEF_AD, previously SU2_DOT_AD)`. The dependency required the build order to be swapped